Docker improvement

fxprunayre opened the pull request:
* fix reverse proxy
* formatting
* synch doc
* fix for port 9300 issue
* fix default dashboard loading
* fix ETF path

fxprunayre committed:
Fix nginx reverse proxy config and formatting file.

fxprunayre committed:
Synch doc and formatting. Ports for tomcat and ES are exposed. Tentative fix for port 9300 issue.

fxprunayre committed:
Merge branch 'docker' of https://github.com/doublebyte1/daobs into docker

doublebyte1 commented on docker/docker-compose.yml:
```diff
@@ -6,9 +6,9 @@ services:
-    #  - "9200:9200"
-    #  - "9300:9300"
+    ports:
+      - "9200:9200"
```
@fxprunayre It does not matter that you publish ports 9200 and 9300 on localhost. the dashboard container will still don't find es on localhost. However, it will find ES on: elasticsearch:9200 and elasticsearch:9300, regardless we publish them.

doublebyte1 commented on docker/elasticsearch/elasticsearch.yml:
```diff
@@ -26,4 +30,4 @@ xpack.graph.enabled: false
 xpack.watcher.enabled: false
 
 #Accept all requests from localhost, dashboard and kibana
-#network.host: [127.0.0.1, localhost, dashboard, kibana, nginx]
```
@fxprunayre the requests from kibana and dashboard are coming from `kibana` and `dashboard`, **not** from `localhost`.

doublebyte1 commented on docker/elasticsearch/elasticsearch.yml:
```diff
@@ -26,4 +30,4 @@ xpack.graph.enabled: false
 
 #Accept all requests from localhost, dashboard and kibana
-#network.host: [127.0.0.1, localhost, dashboard, kibana, nginx]
+network.host: [127.0.0.1, localhost]
```
@fxprunayre please note that the requests to elasticsearch are coming from `dashboard` and `kibana` addresses, **not** from localhost.

fxprunayre committed:
Use docker container name to connect.

fxprunayre committed:
Configure transport address.

fxprunayre committed:
Fix ETF install path.

doublebyte1 committed:
Don't publish ES ports

I remove the publishing of these ports, as it does not make any difference for the dashboard application. You can uncomment locally them for debug purposes.

doublebyte1 committed:
Listening to container hosts

For the application to work, it is more important to allow requests from dashboard, kibana an nginx (?), than localhost, which may only be used for test purposes.

doublebyte1 reviewed (approved):
I reverted a few changes which are not relevant.  Haven't try the others;

fxprunayre committed:
Properly load default dashboard from source folder instead of remote repo. Load them instead of backuping.

fxprunayre committed:
Clean ES config.

fxprunayre committed:
Merge remote-tracking branch 'fxp/docker' into docker

fxprunayre committed:
Merge remote-tracking branch 'joana/docker' into docker

fxprunayre committed:
Fix style.

doublebyte1 committed:
Merge branch 'docker' into docker
